### ISSUE TITLE
Fix browser tab name representation of files with longer than 15 character path

### ISSUE THREAD

**Issue by SarunasAzna** [jupyterlab/jupyterlab#10902]
Tab renaming which was introduced in 3.1.X version shows only [first 15 characters](https://github.com/jupyterlab/jupyterlab/blob/386b5758bbf4d56f08786df9c3f4e5bda14563f5/packages/apputils-extension/src/index.ts#L344) of the file full name. Which can end up showing only a tiny fraction of the path without a file name.
![image](https://user-images.githubusercontent.com/10515132/130612219-13be01cf-ad77-449e-b235-28bfdb0fa4e5.png)
 On the given screenshot full name of the file is `~/path/to/my/file/which/is/deep/name_is_important.ipynb` which is undoubtedly too long for the browser name. Problem is that currently, the browser tab name contains only a fraction of the path without a trace of the filename: `path/to/file - JupyterLab`. So in this case that additional information becomes more or less useless since on the browser with multiple browser tabs opened you can see neither the filename nor the name of the platform (JupyterLab). 

The suggestion would be in this case to remove the path entirely and show only the filename. So the tab name should be looking something like this `name_is_impo... - JupyterLab`. (Or remove filename entirely, since you can see it on JupyterLab tabs)
 If someone has better ideas - please share :)

**Comment by krassowski** [comment]
I agree that full path can be hidden unless if there are two notebooks of the same name (e.g. two Untitled.ipynb) open at the same time, in which case it would be actually useful.

Also, as mentioned in https://github.com/jupyterlab/jupyterlab/issues/10820#issuecomment-897887893 I would very much suport making all of this user-customizable. I don't think anyone committed to working on this, so contributions welcome!

**Comment by tejasmorkar** [comment]
Hello @krassowski, I would like to work on this issue if this is still relevant and up for grabs. 

As I can understand from the mentioned statements, the tab name needs to be updated from the first 15 letters to just the file name of the current notebook truncated and JupyterLab appended to it. This will provide better context than just seeing the path. I can start working on it immediately if I can be assigned to this issue. 

Making it user-customizable also seems a better way of addressing this and I would like to know a little more about how to give options to the user if you already have something in mind.

**Comment by krassowski** [comment]
Thank you, assigned! As for how to make it customisable, it deserves a bit more discussion. We could explore using a template string as a setting, for example:

```json
{
    "windowTitle": {
        "type": "string",
        "description": "The title of the browser window, can use the following pre-defined variables: `workspace`, `file_name`, 'file_path`, `app_name`, `count`",
        "default": "{workspace} {file_name} - {app_name} ({count})"
    }
}
```

The problem is that we do not want to show `{count}` if it is zero or one, and do not want to show workspace if it is the default workspace. I see four possible solutions:
- a) have multiple settings like `windowTitle`, `windowTitleInWorkspace`, `windowTitleManyOpen`, `windowTitleInWorkspaceManyOpen`.
- b) add some kind of conditional logic in the template stings, but ideally we would keep them as simple as possible
    - we do not want to open the can of warms of arbitrary code execution via settings schema so we definitely should try to avoid just using the ES6 template literals and instead use a custom implementation (which would be simple search-and-replace given that there is only a few variables)
- c) have the default point to "{auto}" which would be using hard-coded logic to only show the `count` and `workspace` if needed
   - this is simpler to implement, but forces users customization to a less intelligent solution, which is not great
- d) define extra variables which are empty when not needed, and then remove extra double spaces after substitution; the variables could include:
   - `non_default_workspace` - empty if the workspace is default
   - `count_if_many_in_parens` - empty if count is zero or one, otherwise equal to `({count})`
   - `count_if_many_in_brackets` - empty if count is zero or one, otherwise equal to `[{count}]`

I personally think that the last option (d) of keeping the templating mechanism free of any logic and adding extra variables is the best one, but lets see what others think.

CC @Carreau, @saulshanabrook, @mlucool and @nehz who suggested/upvoted the idea of having template strings and @goanpeca who suggested a few more variables in https://github.com/jupyterlab/jupyterlab/issues/6946#issuecomment-675804557.

**Comment by tejasmorkar** [comment]
Hi @krassowski, I've created PR #10952 for the issue and changed the tab name to a truncated file name as suggested by @SarunasAzna. This will be helpful as the starting 15 characters of the path did not provide any important information. 

If there are multiple files open with the same file names, the path can be shown to differentiate as you said, but should it be added to the tab name? Because personally what I think is, it'll be too long in most of the cases and it'd make more meaning to add the path+file name to the JupyterLab tabs instead of the browser tab. The path is already visible in the URL and as a fullscreen mode is being planned for JupyterLabs, the browser tab will not be visible and instead, JupyterLab tabs will be visible in the fullscreen mode. 

Also, waiting for further discussion on user-customizable options and once a direction is decided I can start working on a PR for that.

Do let me know about the points mentioned and how should it be carried forward.
Thank you.

**Comment by MMesch** [comment]
The triaging agent has identified this issue as likely resolved. See screenshot and rationale below and please reopen and comment in case this is incorrect.

<img width="795" height="257" alt="Image" src="https://github.com/user-attachments/assets/4a59d880-d094-4350-902a-cc75935a99fd" />

---

**Agent Rationale**: The issue has been resolved. PR #10952 was successfully merged implementing exactly what was requested - browser tab titles now show truncated filenames instead of path prefixes for long file paths. The implementation includes proper truncation (12 characters + ellipsis for names > 15 chars) and has been tested. The fix was also backported to the 3.2.x release branch, ensuring it's available to users. The current codebase shows the implemented solution in https://github.com/jupyterlab/jupyterlab/blob/d0efddc366f21752a96b472e4fb1f8dc574d6fd8/packages/apputils-extension/src/index.ts#L402-L408
